registry-plugin/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// pluginName is the plugin name
+var pluginName = "registry"
+
+// HandlerRegisterer is the symbol the plugin loader will try to load. It must implement the Registerer interface
+var HandlerRegisterer = registerer(pluginName)
+
+type registerer string
+
+func (r registerer) RegisterHandlers(f func(
+	name string,
+	handler func(context.Context, map[string]interface{}, http.Handler) (http.Handler, error),
+)) {
+	f(string(r), r.registerHandlers)
+}
+
+func (r registerer) registerHandlers(_ context.Context, extra map[string]interface{}, h http.Handler) (http.Handler, error) {
+
+	config, ok := extra[pluginName].(map[string]interface{})
+	if !ok {
+		return h, errors.New("configuration not found")
+	}
+
+	hostport, _ := config["hostport"].(string)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// If the URL path contains /v2/, indicating a request to Distribution Registry HTTP API V2,
+		// the traffic is hijacked and directed to the registry
+		if strings.Contains(req.URL.Path, "/v2/") {
```

Comment:
I understand this will apply only to the endpoints of a given registry, right? E.g. **registry:5000/artifact/model**/v2.

At first sight applying the same rule to everything that starts by `/v2` seems too broad and likely to cause conflicts in the future but if we consider `<registry-host>/artifact/model` as the model registry (implementing only the Distribution Registry API v2) it will work.

Reply:
Agree to set a registry [prefix](https://distribution.github.io/distribution/about/configuration/#http) to refrain the scope. However, we might set `prefix: /artifact/` instead of `prefix: /artifact/model/`, considering we will have more OCI artifacts to store in the future.